platform/mbed_watchdog_mgr.cpp

```diff
@@ -0,0 +1,108 @@
+
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifdef DEVICE_WATCHDOG
+
+#include "mbed_watchdog_mgr.h"
+
+static bool is_watchdog_started = false; //boolean to control watchdog start and stop
+#define MS_TO_US(x) ((x) * 1000) //macro to convert millisecond to microsecond
+static uint32_t elapsed_ms = (HW_WATCHDOG_TIMEOUT / 2);
+MBED_STATIC_ASSERT((HW_WATCHDOG_TIMEOUT > 0), "Timeout must be greater than zero");
+
+/** Create singleton instance of LowPowerTicker for watchdog periodic call back of kick.
+ */
+static SingletonPtr<mbed::LowPowerTicker> get_ticker()
```

Comment:
Targets which don't support LP TICKER can't support WATCHDOG API ???

Reply:
In the new design watchdog divided into Hw Watcdog and SW watchdog.
1.Hw watchdog registered to low power ticker and expected to call periodically register HW watchdog kick and intern calls HAL Kick API & call process of SW watchdog.(refer mbed_watchdog_mgr.cpp,mbed_wdog_manager_kick() function)
2.SW watchdog has internal mechanism to monitor register service(refer Watchdog.cpp for more info)